### ISSUE TITLE
nearest-exact upsampling returns plain-nearest results: the `+ 0.5` is missing in three kernels

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5228]
## Summary

PyTorch has two nearest-neighbour resize laws, differing only by a half-pixel offset:

```
nearest        src = min(floor( dst        * (IN / OUT)), IN - 1)
nearest-exact  src = min(floor((dst + 0.5) * (IN / OUT)), IN - 1)
```

Three kernels registered for the `_upsample_nearest_exact*` ops use the first formula: `_upsample_nearest_exact1d_kernel`, `_upsample_nearest_exact2d_kernel`, and `_upsample_nearest_exact2d_backward_kernel`. So under `use_gems()`, `mode="nearest-exact"` silently behaves like `mode="nearest"` — every output pixel copies a neighbouring input pixel, and the backward routes each gradient to a neighbouring input cell.

The two laws agree for every integer ratio, and integer ratios are the only ones the tests cover, so the suite sits entirely inside the blind spot.

The kernels were added in #1912, #3866, and #3787.

## Reproduction

```python
import torch
import torch.nn.functional as F

import flag_gems

x = torch.arange(4, device="cuda", dtype=torch.float32).reshape(1, 1, 4)
ref = F.interpolate(x, size=6, mode="nearest-exact")
with flag_gems.use_gems():
    res = F.interpolate(x, size=6, mode="nearest-exact")
print("1d 4 -> 6")
print("  eager:", ref.flatten().tolist())
print("  gems :", res.flatten().tolist())

img = torch.randn(2, 3, 12, 12, device="cuda")
ref = torch.ops.aten._upsample_nearest_exact2d(img, [16, 16], None, None)
with flag_gems.use_gems():
    res = torch.ops.aten._upsample_nearest_exact2d(img, [16, 16], None, None)
print(f"2d (2,3,12,12) -> (16,16): mismatched {(res != ref).sum().item()}/{ref.numel()}")

go = torch.ones(1, 1, 4, 4, device="cuda")
ref = torch.ops.aten._upsample_nearest_exact2d_backward.default(go, (4, 4), (1, 1, 3, 3))
with flag_gems.use_gems():
    res = torch.ops.aten._upsample_nearest_exact2d_backward.default(
        go, (4, 4), (1, 1, 3, 3)
    )
print("2d backward (1,1,3,3) <- (4,4), grad = ones")
print("  eager:", ref.flatten().tolist())
print("  gems :", res.flatten().tolist())
```

```
1d 4 -> 6
  eager: [0.0, 1.0, 1.0, 2.0, 3.0, 3.0]
  gems : [0.0, 0.0, 1.0, 2.0, 2.0, 3.0]
2d (2,3,12,12) -> (16,16): mismatched 672/1536
2d backward (1,1,3,3) <- (4,4), grad = ones
  eager: [1.0, 2.0, 1.0, 2.0, 4.0, 2.0, 1.0, 2.0, 1.0]
  gems : [4.0, 2.0, 2.0, 2.0, 1.0, 1.0, 2.0, 1.0, 1.0]
```

The gems 1-D row is exactly what `mode="nearest"` returns.

## Environment

- FlagGems master (75cf3ea)
- NVIDIA L40S and H100 NVL
- torch 2.6.0+cu124, triton 3.2.0
